tests/system/python/packages/test_lab.py

```diff
@@ -0,0 +1,548 @@
+                    return
+            retries -= 1
+
+        if retries == 0:
+            assert "TIMEOUT! event engine is not running." + foglamp_url + "/foglamp/service"
+
+    # Add Notification with Threshold Rule and Asset Notification (Positive Sine)
+    def test_positive_sine_notification(self, foglamp_url, retries, wait_time):
+        time.sleep(wait_time * 2)
+        data = {"name": "Positive Sine", "description": "Positive Sine notification instance", "rule": "Threshold",
+                "channel": "asset", "notification_type": "retriggered", "enabled": True}
+        post_url = "/foglamp/notification"
+        post_request(foglamp_url, post_url, data)
+
+        data = {"asset": "sinusoid", "datapoint": "sinusoid"}
+        put_url = "/foglamp/category/rulePositive%20Sine"
+        put_request(foglamp_url, put_url, data)
+
+        data = {"asset": "positive_sine", "description": "positive", "enable": "true"}
+        put_url = "/foglamp/category/deliveryPositive%20Sine"
+        put_request(foglamp_url, put_url, data)
+
+        time.sleep(wait_time)
+
+        while retries:
+            get_url = "/foglamp/asset/positive_sine?seconds=600"
+            data = get_request(foglamp_url, get_url)
+            if len(data) > 0:
+                if "event" in data[0]["reading"] and "rule" in data[0]["reading"]:
+                    assert data[0]["reading"]["event"] == "triggered"
+                    assert data[0]["reading"]["rule"] == "Positive Sine"
+                    break
+            retries -= 1
+
+        if retries == 0:
+            assert "TIMEOUT! positive_sine event not fired." + foglamp_url + "/foglamp/asset/positive_sine?seconds=600"
+
+    def test_negative_sine_notification(self, foglamp_url, remove_data_file, retries, wait_time):
+        remove_data_file("/tmp/out")
+
+        payload = {"name": "Negative Sine", "description": "Negative Sine notification instance", "rule": "Threshold",
+                   "channel": "python35", "notification_type": "retriggered", "enabled": True}
+        post_request(foglamp_url, "/foglamp/notification", payload)
+
+        # Upload Python Script (write_out.py)
+        url = foglamp_url + '/foglamp/category/deliveryNegative%20Sine/script/upload'
+        script_path = 'script=@scripts/write_out.py'
+        upload_script = "curl -sX POST '{}' -F '{}'".format(url, script_path)
+        exit_code = os.system(upload_script)
+        assert exit_code == 0
+
+        payload = {"asset": "sinusoid", "datapoint": "sinusoid", "condition": "<"}
+        put_request(foglamp_url, "/foglamp/category/ruleNegative%20Sine", payload)
+
+        payload = {"enable": "true"}
+        put_request(foglamp_url, "/foglamp/category/deliveryNegative%20Sine", payload)
+
+        for _ in range(retries):
+            if os.path.exists("/tmp/out"):
+                break
+            time.sleep(wait_time*2)
+            print("sleeping, to retry the existence check of '/tmp/out'")
```

Comment:
s = wait_time*2
    time.sleep(s)
    print("sleeping for {} before retrying the existence check of '/tmp/out'".format(s))